docs/xapi/mapping/problems.md

```diff
@@ -0,0 +1,196 @@
+| definition["type"]                                                    | http://adlnet.gov/expapi/activities/question                                                                |
+| definition["description"]                                             | <data["submission"][0]["question"]> if <data["submission"][0]["response_type"]> is not empty                |
+| definition["interactionType"]                                         | Mapping of <data["submission"][0]["response_type"]> if <data["submission"][0][response_type"]> is not empty |
+| Context                                                               |                                                                                                             |
+| contextActivities["parent["objectType"]"]                             | Activity                                                                                                    |
+| contextActivities["parent["id"]"]                                     | <LMS_ROOT_URL>/course/<data["course_id"]>                                                                   |
+| contextActivities["parent["definition["type"]"]"]                     | http://adlnet.gov/expapi/activities/course                                                                  |
+| contextActivities["parent["definition["name"]["en-US"]"]"]            | <name of course-run>                                                                                        |
+| extensions["https://w3id.org/xapi/cmi5/context/extensions/sessionid"] | context["session"]                                                                                          |
+| Results                                                               |                                                                                                             |
+| success                                                               | TRUE if <data["success"]>=="correct" else FALSE                                                             |
+| score["min"]                                                          | 0                                                                                                           |
+| score["max"]                                                          | <data["max_grade"]>                                                                                         |
+| score["raw"]                                                          | <data["grade"]>                                                                                             |
+| score["scaled"]                                                       | <data["grade"]>/<data["max_grade"]>                                                                         |
+| response                                                              | <data["submission"][0]["answer"]> if <data["submission"][0]["response_type"]> is not empty                  |
```

Comment:
The `Results["response"]` xAPI field should be a string. However,  the edX `data["submission"][0]["answer"]` seems to be a list (i guess a student can submit multiple answers to different questions in one `problem_check` event).

Would it be possible to move this information to something like `results.extensions['http://vocab.xapi.fr/extensions/responses']` for example?

Alternatively, we could split the `problem_check` event by answer and create one xAPI event for each answer? However, this will likely increase the number of generated xAPI events(

WDYT?

Reply:
https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#interaction-types
I think we are raising a huge question here. We need to take into account the definition["InteractionType"]. Why not creating sub statements (see later in the documentation)?

Reply:
@SergioSim about the problem you raised for a list of responses, we have to use this response pattern if we want to transform a list of multiple answer into a string
https://github.com/adlnet/xAPI-Spec/blob/master/xAPI-Data.md#response-patterns
We can then use the response field rather than an extension and here it goes !!

WDYT?

Reply:
That's a great solution for the case when a student provides multiple answers to a single question (e.g., selects multiple choices).
However, how do we distinguish between multiple answers to multiple questions?
Having a second look at this issue, wouldn't it be easier to have one xAPI "question answered" event by `data.submission` in the `problem_check` event?
We could link the events via a common `context.contextActivities.parent` if there are multiple questions.
WDYT?

Reply:
That was the idea I am thinking of too, otherwise we won't be able to track easily a problem with multiple questions